### DISCUSSION TITLE
[Core Protocol] Why does the first block of the epoch include all the L2 deposits that happened in the L1 block?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [ethereum-optimism/developers#416] [Core Protocol] [answered]
### Did you check the documentation?

- [X] Yes

### Did you read the specs?

- [X] Yes
- [ ] No

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

From the [docs](https://docs.optimism.io/stack/protocol/rollup/overview#block-production):

> Transactions get to the sequencer in two ways:
>
> 1. Transactions submitted on L1 (called deposits) are included in the chain in the appropriate L2 block. Every L2 block is identified by the "epoch" (the L1 block to which it corresponds, which typically has happened a few minutes before the L2 block) and its serial number within that epoch. **The first block of the epoch includes all the deposits that happened in the L1 block to which it corresponds.**

Why is that?

I might have misunderstood this part of the docs, but I would have expected the first block of an epoch to contain all the deposits of the _previous epoch_.

Doesn't this refer to the [epochs on Ethereum Mainnet](https://beaconcha.in/epochs)?

**Comment by imtipi** [comment, marked as answer]
> Doesn't this refer to the [epochs on Ethereum Mainnet](https://beaconcha.in/epochs)?

> Every L2 block is identified by the "epoch" (the L1 block to which it corresponds, which typically has happened a few minutes before the L2 block) and its serial number within that epoch

it is L2's epoch, a L1 block

**Reply by PaulRBerg** [reply]
Thank you for your answer.

In this case, I highly, highly recommend updating the docs to specify that this is NOT the same as the Beacon Chain epochs.

**Reply by PaulRBerg** [reply]
https://github.com/ethereum-optimism/docs/issues/703

**Reply by imtipi** [reply]
here's some docs for you https://specs.optimism.io/protocol/overview.html?highlight=epoch#block-derivation
The L2 epochs are explained more concisely here.

> The rollup chain is subdivided into epochs. There is a 1:1 correspondence between L1 block numbers and epoch numbers.

**Reply by PaulRBerg** [reply]
Thanks, that's a helpful reference.

I think I understand it now:

- L1 blocks are produced every 12 seconds (on average)
- L2 blocks are produced every 2 seconds (on average)
- Thus, on average, six L2 blocks will be associated with one L1 block (the 'epoch')
- Epoch $i$ corresponds to L1 block $i$ which corresponds to L2 blocks $j$, $j+1$, $j+2$, $j+3$, $j+4$, and $j+5$
- It only L2 block $j$ that contains L1 deposit txs because the rest are all linked to the same L1 block
